### PR TITLE
add OpenMage admin theme and theme switcher

### PR DESCRIPTION
**Pull request by aterjung**
in addition to logo integration build a draft version of a new styled backend-theme for OpenMage.  I think it is more and more important to show the ongoing development to the users. I would like this to be seen as a first step - a new Backend-Style mainly build by css changes on top of the old default/default admin theme to be as compatible as possible. I my mind, there should follow a second step, with a more up-to-date approach.

This PR consists of to parts, the theme switcher to switch from OpenMage Theme to the legacy Magento M1 theme, and the theme itself.

- For the theme switcher, is accepted to build it into the core, or should i build it as a new module?

The theme itself supply only one phtml file to change the login, because that can not be done via layout files. For the themes CSS i used some parts from https://github.com/jreinke/magento-admin-theme and converted them to SCSS. I have replaces as many icons as possible to inline-SVGs.

Help with the styling would be very welcome!

### CODE REVIEW

**Comment by Schrank** [comment]
Would you mind to add a screenshot? I would love to get an impression without installing it 😅

**Comment by aterjung** [comment]
no problem, here are some screens!
Bildschirmfoto 2020-05-27 um 21 10 42
Bildschirmfoto 2020-05-27 um 21 11 43
Bildschirmfoto 2020-05-27 um 21 12 22
Bildschirmfoto 2020-05-27 um 21 12 56
Bildschirmfoto 2020-05-27 um 21 13 09

**Comment by aterjung** [comment]
what i did not ask in my comment, is this a change for the 19.4 branch, because it is a non breaking change, or should it go to the 20.4 branch because is something new? At the 20.4 branch, i don't see progress at the moment?

**Comment by colinmollenhour** [comment]
Looking great!

Please make sure you include somewhere instructions on generating the CSS files from the SCSS files so that anyone may easily add updates and not have to figure out the tooling from scratch. A comment in the generated CSS file to prevent users from directly editing them would be good to avoid confusion.

**Comment by colinmollenhour** [comment]
Forgot to address the branch question.. You have it on the right branch. How the version is tagged after this is merged I think I would defer to @Flyingmana.

Upon upgrade it would be nice if this is the default skin as long as users can go back to the old one in case it breaks their extensions somehow.

**Comment by Flyingmana** [comment]
if there is no BC break, adding it to 1.9.4.x sounds good enough.


Regarding 20.x , I need to add a note to merge the changes upstream and also do a release there <.<

**Comment by aterjung** [comment]
i have added a comment to prevent editing the css files. Also i have replaces a lot of old looking icons with svg versions - but still everything via css. From my point of view, that should be BC, but  it has to be tested with the usual extensions.

@colinmollenhour its allready build that way, the OpenMage theme is as standard theme, but the old legacy theme can be selected from backend.

**Comment by sreichel** [comment]
GREAT work!

Whats about error/debug pages? (https://github.com/OpenMage/magento-lts/tree/1.9.4.x/errors/default)

**Comment by colinmollenhour** [comment]
Any reasons not to go ahead and merge this? I loaded it up and tested it out and everything works well.

**Comment by luigifab** [comment]
Is the theme can be set per admin user? (in System / My account).

**Comment by aterjung** [comment]
i tested it myself in some customer installations without problems. From my side, it would be ready to merge. I will change the state!

**Comment by aterjung** [comment]
I have updated the color theme a bit to reflect the upcoming new website. I hope you like it!
Only Color-Values changed, def. no BC!
Bildschirmfoto 2020-06-11 um 00 24 53

File: app/code/core/Mage/Adminhtml/Controller/Action.php [hunk 1]
```diff
@@ -151,12 +151,12 @@ public function preDispatch()
     {
         // override admin store design settings via stores section
 
-        // get theme form backend config
         $legacy = Mage::getStoreConfig('admin/design/theme');
```
Comment (sreichel) [inline app/code/core/Mage/Adminhtml/Controller/Action.php]: Not needed anymore ;)

Comment (aterjung) [reply app/code/core/Mage/Adminhtml/Controller/Action.php]: followed your Suggestion now! Thinking about it today, it's the right way.

**Comment by sreichel** [comment]
I'll test on saturday!

There is no default config value for `Mage::getStoreConfigFlag('admin/design/theme')` .

```
    <default>
        <admin>
            <design>
                <theme>0</theme>
```

.... and maybe rename it to `use_legacy_theme` to make it more clear what the option is for.

**Comment by aterjung** [comment]
> There is no default config value for `Mage::getStoreConfigFlag('admin/design/theme')` .
> 
> ```
>     <default>
>         <admin>
>             <design>
>                 <theme>0</theme>
> ```
> 
> .... and maybe rename it to `use_legacy_theme` to make it more clear what the option is for.

both changes are added now!

File: app/code/core/Mage/Core/etc/system.xml [hunk 2]
```diff
@@ -1058,6 +1058,25 @@
...
+                    <show_in_website>0</show_in_website>
+                    <show_in_store>0</show_in_store>
+                    <fields>
+                        <use_legacy_theme translate="label comment">
+                            <label>Enable legacy Admin Theme</label>
+                            <comment>Use legacy theme from Magento CE</comment>
+                            <frontend_type>select</frontend_type>
+                            <source_model>adminhtml/system_config_source_yesno</source_model>
+                            <sort_order>1</sort_order>
+                            <show_in_default>1</show_in_default>
```
Comment (luigifab) [inline app/code/core/Mage/Core/etc/system.xml:1075]: can I suggest to add also show_in_website and show_in_store even if the parent is not viewable in website and store?

**Comment by sreichel** [comment]
LGT. Found just some missing header-icons. :100: 

And maybe check spacing for dropdowns .... Reports -> Customers -> New customers .... "Show by Day" looks like "Show by Dav".

File: app/code/core/Mage/Core/etc/system.xml (hunk 2 above)
Comment (aterjung) [reply app/code/core/Mage/Core/etc/system.xml:1075]: > LGT. Found just some missing header-icons. 💯

What means LGT? Could you name where the missing icons should be? I only see some missing icons that were missing in a vanilla magento as well. Like on "Catalog Price Rules"

> And maybe check spacing for dropdowns .... Reports -> Customers -> New customers .... "Show by Day" looks like "Show by Dav".

Fixed!

Thanks for testing!

**Comment by aterjung** [comment]
> LGT. Found just some missing header-icons. 💯

What means LGT? Could you name where the missing icons should be? I only see some missing icons that were missing in a vanilla magento as well. Like on "Catalog Price Rules"

> And maybe check spacing for dropdowns .... Reports -> Customers -> New customers .... "Show by Day" looks like "Show by Dav".

Fixed!

Thanks for testing!

File: app/code/core/Mage/Core/etc/system.xml (hunk 2 above)
Comment (aterjung) [reply app/code/core/Mage/Core/etc/system.xml:1075]: just commited your suggestion!

**Comment by sreichel** [comment]
> What means LGT? Could you name where the missing icons should be? I only see some missing icons that were missing in a vanilla magento as well. Like on "Catalog Price Rules"

Missing M ... LGTM :+1: 

If they are missing in vanilla too its okay .... maybe want to add some, but nothing whats blocks this :)

**Comment by aterjung** [comment]
> Missing M ... LGTM 👍
;-)

> Have ju updated PR?
damn, i forgot to push the changes. Done now!

**Comment by colinmollenhour** [comment]
@aterjung, although the theme that you based this on has been public for six years, I think to be safe it would be good to ask the author to add a license to it (e.g. AFL 3.0 like other Magneto themes) or otherwise "bless" the contribution of his theme into OpenMage.

**Comment by aterjung** [comment]
I have written an email to jreinke two weeks ago, but im not shure if the e-mail address was right - i don't get any response. I tried it now again with a different address i found on his git commits log. 
Only statement from him i could find about this is from 2013: https://github.com/jreinke/magento-admin-theme/issues/9 It sounds like he don't have a problem with sharing his code - but that is realy old.
I am not to shure about how the AFL 3.0 is handled in this case - his work is based on the magento Admin theme itself - but it looks to me as if the AFL lets you choose a different licence for derivative work?
If we don't get a written permission - it would than be needed to remove alle parts made from him and rewrite it - that would be possible - but would take 2-3 Days in full time i think.

**Comment by colinmollenhour** [comment]
Agreed that there is almost certainly not going to be an objection from him based on his previous statement, but I think we should get a response before merging. I submitted a PR: https://github.com/jreinke/magento-admin-theme/pull/24

File: app/design/adminhtml/default/openmage/layout/local.xml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0"?>
+<layout version="0.1.0">
+    <default>
+        <reference name="head">
+            <action method="addLinkRel"><rel>stylesheet</rel><href>//fonts.googleapis.com/css?family=Open+Sans:400,300,600,700</href></action>
```
Comment (kkrieger85) [inline app/design/adminhtml/default/openmage/layout/local.xml:5]: Why is it necessary to load fonts from remote?

Comment (aterjung) [reply app/design/adminhtml/default/openmage/layout/local.xml:5]: good point, i forgot to address this - was somewhere on my todo list. The Open Sans Font could be dropped, or alternatively - what i would prefer - distributed with openmage. It is licensed under the Apache License which permits distribution as long as a copy of the license is packed along.
I like the lighter font weight that is possible with this font, so i would not drop it to fall back to Arial...

Comment (aterjung) [reply app/design/adminhtml/default/openmage/layout/local.xml]: i have added the Open Sans Font as local font to theme and removed the link to google fonts. I hope that embedding the font to OpenMage would be acceptable. I have added the license.txt for the font too.

**Comment by aterjung** [comment]
> Agreed that there is almost certainly not going to be an objection from him based on his previous statement, but I think we should get a response before merging. I submitted a PR: [jreinke/magento-admin-theme#24](https://github.com/jreinke/magento-admin-theme/pull/24)

Until now, now reaction ;-( I will give him some more days, than i will strip out all of his parts and start to rebuild myself.

**Comment by jhejlik** [comment]
Hi guys,
I'm afraid this switcher Enable legacy (yes/no) do not allow to use other custom admin themes.
Would it be better to use only a text field with the theme name is entered like this switcher which I use on many stores?

https://github.com/jhejlik/Mageek_OpenmageTheme

theme-switcher

**Comment by Schrank** [comment]
Dropdown to make it easier?

**Comment by sreichel** [comment]
> 
> Dropdown to make it easier?

:+1: 

@jhejlik can you add a node to your `config.xml`?

```
</config>
    <stores>
        <admin>
            <design>
                <theme>
                    <YOUR_THEME>some name</YOUR_THEME>
                </theme>
            </design>
        </admin>
    </stores>
</config>
```

I'd should be easy to make a dropdown via ...

```
$themes = Mage::getConfig()->getNode("stores/admin/design/theme");
$themes->asArray();
```

**Comment by jhejlik** [comment]
> Dropdown to make it easier?

@Schrank dropdown with themes will be better

> can you add a node to your ? `config.xml`

@sreichel yes i can, great 👍

**Comment by sreichel** [comment]
> @sreichel yes i can, great +1

Can you please add a PR to @aterjung project? Add config section, with dropdown, ...?

**Comment by aterjung** [comment]
@jhejlik did you build this from scratch on your own? Would you mind adding a licence that make it possible to integrate your theme or parts of it to OpenMage?

To the original question: Did you tested your theme with my admin theme branch? I did not test it, but if i am not totaly wrong it should not conflict with your theme switch.

**Comment by jhejlik** [comment]
@aterjung I have not been testing it, it is my fault. I will test it tonight.

Yes, the theme is my work only. The theme switcher is based on the 8-year-old [Inchoo_AdminTheme](https://github.com/ajzele/Inchoo_AdminTheme), but I assume it will not be used for eventual integration. What license do you mean (need)? MIT, GPL or other?

**Comment by aterjung** [comment]
right, the theme switcher is not needed. I think the AFL 3.0 License used in all magento own themes would be the best.

I like the smaller Header you build - did you have any problems with extensions with this?

Did you used SASS or LESS for your css development? If yes, would you supply the source files?

I will try out your theme - probably it could help me a lot getting rid of the css from jreinke. What should be discussed here, should we provide more than one new admin theme if possible - or should we provide the infrastructure to integrate these as additional modules?

**Comment by jhejlik** [comment]
@aterjung I tested your branch with my theme switcher and all working fine without errors. So i apologize for my concern.

In my template i did not use any preprocessor. My css file overwrites and extends the default magento styles only, not replacing them. I know that's not a good solution, there is more code.  I plan to overwrite the original css files to make them more thin. And I will have to edit a few phtml files due to the old gifs inserted. 
I have been using this theme for about 3 years and no incompatibility, just a few trifles. But there are extensions where developers have rewritten layouts and styles with their own. Compatibility cannot be ensured there.

**Comment by aterjung** [comment]
> In my template i did not use any preprocessor. My css file overwrites and extends the default magento styles only, not replacing them. I know that's not a good solution, there is more code. I plan to overwrite the original css files to make them more thin. And I will have to edit a few phtml files due to the old gifs inserted.

I think that is not the wrong way - indeed the css files are bigger than needed and often it would be more easy to start from scratch without these masive load of historical garbage - but i am scared that some extensions rely on these definitions. But it would be intresting to see if that will realy happen.

What do you think about the license topic - would you contribute your work to OpenMage? I would merge in some changes i did on my first Theme attempt and adjust the color theme.

**Comment by jhejlik** [comment]
> What do you think about the license topic - would you contribute your work to OpenMage

@aterjung I can't add an afl license to the repository when it contains code with a different rights. How do i do that? Should i insert the license directly into the css files only?

**Comment by aterjung** [comment]
@jhejlik i would say, add the license file to the repo and mention the AFL in the css file would be good!

**Comment by jhejlik** [comment]
@aterjung license updated

**Comment by aterjung** [comment]
@jhejlik great, thank you!

Question to all, what do you think, i would like to use jhejlik`s theme as base to rebuild the OpenMage theme just like my first version - but without the legal concerns. I like the smaller header and the Login from this new theme - but i have concerns that these changes could break the compatibility with some extensions cause they need changes to the template code - not just css. So should we add more then one admin theme - lets say Legacy/OpenMage Standard/OpenMage compact?

**Comment by jhejlik** [comment]
@aterjung I still think it is better not to limit the settings to legacy or openmage only. There are agencies and stores that use their own rebranding in administration.
Dropdown could contain an automatic list of theme folders in the adminhtml/default package directory, but text input would be enough as for frontend theme settings, it is not something that changes everyday.

**Comment by colinmollenhour** [comment]
Just so you know, I'm ok with either theme, they both look good. The @aterjung theme is probably less likely to have BC issues, but the @jhejlik theme header looks more unique. I'm not too keen on supporting both themes but it is so hard to decide. :)

As far as the licensing issue, I have not been able to get any response from Johann yet... It seems pretty clearly public domain give the age and the previous statements so I think we could merge it if that is the preferred theme.

How can we come to a decision here so we can get one or the other merged? What are the blockers and which theme does everyone prefer? Perhaps poll the Discord?

**Comment by sreichel** [comment]
> Question to all, what do you think, i would like to use jhejlik`s theme as base to rebuild the OpenMage theme just like my first version - but without the legal concerns. I like the smaller header and the Login from this new theme

Yes, the smaller header is nice (maybe just smaller logo?), but i like your coloring bit more. I like the SCSS way, too .... so I'd prefer yours. You started this here  so it would be just fair. I like both. Supporting both is some overhead ... it's hard, but it would be okay for me.

Blockers? A dropdown template switscher would be nice. I'd prefer to have to add a xml node to register custom admin theme.

**Comment by colinmollenhour** [comment]
Just FYI, Johann merged the license file!

**Comment by aterjung** [comment]
> Just FYI, Johann merged the license file!

 great news! But the decision is not easier now...

If you give me some days - in germany we have a corona vat change in July 1. in addition to the EOL that keeps me busy - i will try to merge the best of both themes. Probably that will be the one to choose...

**Comment by sreichel** [comment]
> If you give me some days - in germany we have a corona vat change in July 1

Please check N98-magerun ... last days they added a command to change taxes ....

**Comment by aterjung** [comment]
some teasing from my work this evening, screens from merging the best of all themes...

Sadly - the scss now needs a lot of cleanup before i can push an update. But some more info to the screens: 
Bildschirmfoto 2020-06-30 um 23 37 43
Ive removed the background loading from unsplash - but selected one background that matches the colors OpenMage quite well. But i don't want to introduce more calls to external servers.

Bildschirmfoto 2020-06-30 um 23 37 31

I like this smaller header a lot, so i adopted the idea to the OpenMage theme.

Now i would start to rework the scss files an do some cleanup.

**Comment by colinmollenhour** [comment]
Looks great! Keep it up, can't wait to merge it!

**Comment by colinmollenhour** [comment]
Once the style is settled I'd like to feature a single screenshot on the homepage to showcase how the admin panel is new and refreshed and unique, while still being recognizable to Magento 1 users. :)

Can you provide a good screenshot for this purpose? Perhaps use some real data on the dashboard but use DOM inspector to change the names and any private details? I'll leave it up to you, anything that you think showcases it well except perhaps for the login.. :)

**Comment by jhejlik** [comment]
You are all great programmers (which i am not) but the final GUI should definitely be considered by an experienced backend designer. At the first glance of screenshots, i see insufficient contrast of the text. Visually impaired users or users with older TN panels will have problems with font readability. Graphic design has its WCAG standards. 

Keep in mind that administration is not website that a user leave in 5 minutes, but an interface in which thousands of users will spend many hours a day. Openmage administration does not need strong colors. The colors pleasant for all-day work have to be chosen. Most importantly, the colors must be compatible with many extensions designing for M1, we have to avoid a hodgepodge. And the font have to be available in all subsets. Not like M2, where local chars do not work.

I do not want to spoil anyone's joy and i do not want to be pessimistic. i am proud you have chosen [my admin theme](https://github.com/jhejlik/Mageek_OpenmageTheme) as inspiration. I have been optimizing it for 3 years and it works very well. 

With all due respect I think there is no time for a complete colors changing of the administration. We must first make it easier for users to switch to Openmage and ensure maximum compatibility. We can consider repainting later after creating our own admin designsystem.

**Comment by aterjung** [comment]
I am open to any optimization, and yes - I am not da designer. But the reason to change the backend look and feel is more a political thing. We urgently need to build our own identity and a big change in the Design shows progress to normal users - they don’t see what’s happened to the codebase - they don’t see hundreds of commits, but they see - hey everything looks new and shiny.

**Comment by Flyingmana** [comment]
@jhejlik thank you for the insights. Indeed its very important to keep in mind the usage of the Admin Area is very different and optimization for long usage as well as Accessibility should have a high priority for this kind of changes.
Every Feedback you can provide in this area is very valuable for us, as not all of us have this kind of experience yet.(I for example only know some basics)

Regarding the Compatibility, this indeed is important, which is why we will keep this optional (configurable) so in case of issues people can easy switch back.
Regarding the Time, we by now have many contributors in different areas, so this is not causing disruptions in other areas (and we prefer to have volunteers working in areas, they feel enthusiastic about. Mostly because they are doing this voluntary and we are very thankfully for every contribution. And no Volunteer should feel forced to invest time into something, they dont want to)

**Comment by colinmollenhour** [comment]
Great work here, everyone and good discussion too. I think a few key points here are:

- We need to deploy a change very soon for reasons stated by others
- The new theme should be the default, but users should be able to easily switch back to the old theme if it causes them problems or they simply don't like it (versus making it optional - nobody would notice it or use it)
- Although it doesn't need to be perfect on first release it should be very good.

**Comment by colinmollenhour** [comment]
Let's please get this merged soon. My designer is working on some really cool graphics for the homepage and it would be great if he's not having to design with a moving target. For example:

image

Let's get a new theme merged and then I'm sure we'll all be happy to accept PRs with UX improvements.

Are there any blockers for merging in the current state?

**Comment by colinmollenhour** [comment]
Can someone supply some nice screenshots of the dashboard with real, but anonymous data?

**Comment by aterjung** [comment]
i think the state is pretty stable at the moment, while a bit ugly from the code view (mainly scss) I think that should not prevent this state to be merge, it is more a point for easying future improvments. For me it would be ok to do this in a second step.
Whats needs to be finally discussed is the switcher. At the moment we have a boolean config value - to make custom theme development easier we could got with a source-model and read the themes from config files?

I will send you some screenshots in the evening! (GMT+2)

**Comment by aterjung** [comment]
i have pushed a new state with all updates now. Can someone test it with some older IEs? I think we should be shure to not break the backend completely for someone with an outdated IE?

@jhejlik i hope i was not to brusque - it would be great profit from your experience in the next steps!

**Comment by aterjung** [comment]
@colinmollenhour is send you screenshots via Discord, did you receive them?

**Comment by colinmollenhour** [comment]
> @colinmollenhour is send you screenshots via Discord, did you receive them?

Yes, thanks!

**Comment by ioweb-gr** [comment]
Just a small question, will the theme be responsive ?

**Comment by colinmollenhour** [comment]
Screenshots were added here: https://www.openmage.org/magento-lts/about.html

We did some testing on an earlier version on a store with tons of custom extensions and compatibility was excellent.

Are there any blockers for merging this now?

File: app/design/adminhtml/default/openmage/template/page/header.phtml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+		    </li>
+		</ul>
+        <?php if ( Mage::getSingleton('admin/session')->isAllowed('admin/global_search') ): ?>
+        <fieldset>
+            <legend>Search</legend>
+            <span id="global_search_indicator" class="autocomplete-indicator" style="display: none">
+                <img src="<?php echo $this->getSkinUrl('images/ajax-loader.gif') ?>" alt="<?php echo Mage::helper('core')->quoteEscape($this->__('Loading...')) ?>" class="v-middle"/>
+            </span>
+            <?php $defSearch = $this->__('Global Record Search') ?>
+            <input id="global_search" name="query" type="text" class="input-text" value="<?php if(!empty($query)): ?><?php echo $query ?><?php else: ?><?php echo Mage::helper('core')->quoteEscape($defSearch) ?><?php endif ?>" onfocus="if(this.value=='<?php echo Mage::helper('core')->quoteEscape($defSearch, true) ?>')this.value=''; " onblur="if(this.value=='')this.value='<?php echo $defSearch ?>';" autocomplete="off" />
```
Comment (colinmollenhour) [inline app/design/adminhtml/default/openmage/template/page/header.phtml:59]: Where is $query expected to be defined?

File: app/design/adminhtml/default/openmage/template/page/header.phtml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        		<a href="#" onclick="return false" title="<?php echo $this->__('User Menu') ?>">
+                    <svg version="1.1" id="burger" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="40px" height="40px" viewBox="0 0 40 40" enable-background="new 0 0 40 40" xml:space="preserve">
+                        <rect opacity="0.25" fill="#FFFFFF" width="40" height="40"></rect>
+                        <path fill="#FFFFFF" d="M31,21H9v-2h22V21z M31,17H9v-2h22V17z M31,25H9v-2h22V25z"></path>
+                    </svg>
+                </a>
+			    <ul>
+			    	<li><span><?php echo $this->formatDate(null, 'medium') ?></span></li>
+                    <li><a href="<?php echo Mage::helper("adminhtml")->getUrl("adminhtml/system_account"); ?>"><span><?php echo $this->__("Logged in as %s", $this->escapeHtml($this->getUser()->getUsername())); ?></span></a></li>
+			    	<li><a href="<?php echo Mage::getBaseUrl();; ?>" target="_blank"><span><?php echo $this->__('View') . ': ' . Mage::app()->getStore()->getFrontendName(); ?></span></a></li>
```
Comment (colinmollenhour) [inline app/design/adminhtml/default/openmage/template/page/header.phtml:47]: Needs `escapeHtml` around store name.

File: app/design/adminhtml/default/openmage/template/resetforgottenpassword.phtml [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                            <input type="password" class="input-text no-display" name="dummy" id="dummy" />
+                            <input type="password" class="input-text required-entry validate-cpassword" name="confirmation" id="confirmation" autocomplete="new-password"/>
+                        </div>
+                        <div class="clear"></div>
+                        <div class="form-buttons">
+                            <a class="left" href="<?php echo $this->getUrl('adminhtml', array('_nosecret' => true)) ?>">&laquo; <?php echo Mage::helper('adminhtml')->__('Back to Login'); ?></a>
+                            <button type="submit" title="<?php echo Mage::helper('core')->quoteEscape(Mage::helper('adminhtml')->__('Reset Password')); ?>" class="forgot-password"><span><span><span><?php echo Mage::helper('adminhtml')->__('Reset Password'); ?></span></span></span></button>
+                        </div>
+                    </fieldset>
+                    <p class="legal"><?php echo Mage::helper('adminhtml')->__('Magento is a trademark of Magento Inc. Copyright &copy; %s Magento Inc.', date('Y')); ?></p>
```
Comment (colinmollenhour) [inline app/design/adminhtml/default/openmage/template/resetforgottenpassword.phtml:74]: We removed all Magento logos and the word Magento so we should be able to also remove this legal notice.

File: app/design/adminhtml/default/openmage/template/forgotpassword.phtml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+                        <input type="text" id="email" name="email" value="" class="required-entry input-text forgot-password validate-email" style="width:461px;" />
+                    </div>
+                    <?php echo $this->getChildHtml('form.additional.info'); ?>
+                    <div class="clear"></div>
+                    <div class="form-buttons">
+                        <a class="left" href="<?php echo $this->getUrl('adminhtml', array('_nosecret' => true)); ?>">&laquo; <?php echo Mage::helper('adminhtml')->__('Back to Login'); ?></a>
+                        <input type="submit" class="form-button" onclick="loginForm.submit()" value="<?php echo Mage::helper('adminhtml')->__('Retrieve Password'); ?>">
+                    </div>
+                </fieldset>
+                <p class="legal"><?php echo Mage::helper('adminhtml')->__('OpenMage - Actively supported fork of Magento Community Edition') ?></p>
```
Comment (colinmollenhour) [inline app/design/adminhtml/default/openmage/template/forgotpassword.phtml:63]: Please just call it OpenMage LTS and do not mention Magento CE. See discussion in https://github.com/OpenMage/magento-lts/pull/1014

File: app/design/adminhtml/default/openmage/template/login.phtml [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+ * @package     default_default
+ * @copyright   Copyright (c) 2006-2018 Magento, Inc. (http://www.magento.com)
+ * @license     http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+ */
+?>
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml" lang="en">
+<head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <title><?php echo Mage::helper('adminhtml')->__('Log into Magento Admin Page') ?></title>
```
Comment (colinmollenhour) [inline app/design/adminhtml/default/openmage/template/login.phtml:31]: Please replace "Magento" with "OpenMage LTS"

File: app/design/adminhtml/default/openmage/template/login.phtml [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+                    <div class="input-box"><label for="login"><?php echo Mage::helper('adminhtml')->__('Password:') ?></label><br />
+                        <input type="password" id="login" name="login[password]" class="required-entry input-text" value="" /></div>
+                    <?php echo $this->getChildHtml('form.additional.info'); ?>
+                    <div class="clear"></div>
+                    <div class="form-buttons">
+                        <a class="left" href="<?php echo Mage::helper('adminhtml')->getUrl('adminhtml/index/forgotpassword', array('_nosecret' => true)) ?>"><?php echo Mage::helper('adminhtml')->__('Forgot your password?') ?></a>
+                        <input type="submit" class="form-button" value="<?php echo Mage::helper('core')->quoteEscape(Mage::helper('adminhtml')->__('Login')) ?>" title="<?php echo Mage::helper('core')->quoteEscape(Mage::helper('adminhtml')->__('Login')) ?>" /></div>
+                </div>
+                <p class="legal"><?php echo Mage::helper('adminhtml')->__('OpenMage - Actively supported fork of Magento Community Edition') ?></p>
```
Comment (colinmollenhour) [inline app/design/adminhtml/default/openmage/template/login.phtml:67]: Please just call it OpenMage LTS and do not mention Magento CE. See discussion in https://github.com/OpenMage/magento-lts/pull/1014

File: skin/adminhtml/default/openmage/login.css [hunk 10]
```diff
@@ -0,0 +1,281 @@
...
+  margin: 120px 0 0;
+  padding-left: 200px;
+  background: url(http://www.localhost.com:3000/skin/adminhtml/default/mageek/images/logo.svg) left center no-repeat;
+  margin-left: calc(50vw - 372px);
+}
+.login-container .legal {
+  margin: 0;
+  background: url(http://www.localhost.com:3000/skin/adminhtml/default/default/images/login_box_legal_bg.gif) no-repeat;
+  padding: 8px 8px 5px 8px;
+  font-size: 0.95em;
```
[the remark this answers is not in the document]
Comment (aterjung) [reply skin/adminhtml/default/openmage/login.css]: thanks for comment, i have fix this now

File: app/design/adminhtml/default/openmage/template/page/header.phtml (hunk 4 above)
Comment (aterjung) [reply app/design/adminhtml/default/openmage/template/page/header.phtml:59]: i don`t know, cant find it at the moment. it is used the same way in the original magento header.phtml file. I am not shure if it is is created anywhere.

**Comment by aterjung** [comment]
as far as i can see, every mentioned issue is solved now? I will add optimisations in the future, but i think it should not block merging now.

**Comment by aterjung** [comment]
anything i should change? I intended to stop any further development until this state is merged?

**Comment by colinmollenhour** [comment]
Great work everyone! Please feel free to submit separate PRs for improvements.

**Comment by ioweb-gr** [comment]
Could this be tagged as a minor release? I'd love to try it

**Comment by colinmollenhour** [comment]
Oops, I didn't realize there was a 20.0 branch.. I just merged it into the 20.0 branch.  @Flyingmana do you want me to revert it from 1.9.4.x? I'll leave the release strategy up to you.

**Comment by sreichel** [comment]
@colinmollenhour  IMHO 1.9.4.x is correct ... (or is it clear what the 20.x branch is for?)

**Comment by sreichel** [comment]
> anything i should change?

After it is merged ... see my comment above ;)

Thanks for your work :100:

**Comment by Flyingmana** [comment]
@colinmollenhour
If it has a potential to cause issues, then 20.0, else its fine as it is.

**Comment by sreichel** [comment]
Backend is fine, but I still see the old admin login page? CSS is not loaded?

**Comment by ADDISON74** [comment]
After 12 years of using M1 it is difficult to ask people to adapt to other backend theme. We should be patience with everyone. This new theme should be a proposal not an imposition. Personally I will advice anyone to test it before using it. It could be a package installed by those who want it.

Related to the style of the theme I agree we need a SCSS file to change easy the values. Many programmers are not UX designers. I suggest a limitation in using colors and please use a Color Wheel to choose correctly the color pallet. You may use Adobe Color Wheel from here: https://color.adobe.com/create/color-wheel. Or ColorImpact from here: https://www.tigercolor.com/. Probably the colors look OK for the eye but the brain could have some issues with the chosen ones. I would like to see a different logo for OpenMage to express an idea not an m with a text.

This is a personal opinion and should be taken as such ...

**Comment by LuizSantos22** [comment]
I would give as a inspiring idea for new admin theme release, the extension created by CMSmart Admin Theme for Magento 1
I used to use it when I was still using "pure" magento. But after upgrading to OpenMage, it has stopped working
and I did not bother trying ways to fix it. 
This is their link: http://preview.cmsmart.net/demo/magento-responsive-admin-template (Magento 1 is on the left)

It has icons on the menu names and it can be set to be horizontal (like normal magento) and vertical (like magento 2). 
In my opinion, vertical menu is more elegant than horizonal one.

**Pull request merged**